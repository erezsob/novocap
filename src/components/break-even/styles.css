.container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 20px;
    width: 90%;
    padding-top: 20px;
    color: white;
    font-weight: bold;
    font-size: 20px;
}

.barContainer {
    margin: 20px;
    width: 150px;
    height: 150px;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.title {
    display: flex;
    flex-direction: column;
    padding-top: 13px;
}

.date {
    font-size: 11px;
    font-weight: 300;
}

.addCosts {
    color: #fff;
    font-size: 19px;
}