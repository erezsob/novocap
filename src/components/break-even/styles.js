import { blueMain } from '../../common/variables.css';

export default {
  card: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    boxShadow: 'none',
    height: 208,
    backgroundColor: blueMain,
    cursor: 'pointer'
  },
  iconButton: {
    color: 'white',
    width: 24
  }
};
