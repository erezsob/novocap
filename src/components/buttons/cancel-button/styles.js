import { pinkyRed } from '../../../common/variables.css';

export default {
  cancel: {
    color: pinkyRed
  }
};
