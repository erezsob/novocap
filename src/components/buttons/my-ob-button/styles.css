@value blueLight, bpXSmall from '../../../common/variables.css';

.button {
    border-radius: 4px;
    background-color: blueLight;
    color: white;
    padding: 8px 0;
    font-size: 12px;
    display: table;
    border: 0;
    outline: 0;
    cursor: pointer;
}

.bigger {
    padding: 10px 0;
    font-size: 16px;
}

.icon,
.text {
    vertical-align: middle;
    text-align: center;
}

.text {
    padding: 0 20px;
    font-weight: lighter;
}

.icon ~ .text {
    display: none;
}

.icon {
    display: table-cell;
    padding: 0 10px;
    position: relative;
    top: 2px;
}

@media (min-width: bpXSmall) and (min-device-width: bpXSmall) {
    .icon ~ .text {
        padding-left: 0;
        display: table-cell;
    }
}
