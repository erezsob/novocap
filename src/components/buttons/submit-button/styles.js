import { greenGrass } from '../../../common/variables.css';

export default {
  submit: {
    color: greenGrass
  }
};
