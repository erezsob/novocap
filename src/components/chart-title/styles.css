.label {
    letter-spacing: 0.05em;
    font-size: 20px;
    border-bottom: 1px solid #efefef;
    padding: 20px;
    margin-bottom: 20px;
    font-weight: bold;
    color: #323435;
}