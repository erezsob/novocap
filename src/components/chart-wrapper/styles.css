.container {
    padding: 0 20px 30px;
    position: relative;
    font-size: 12px;
}