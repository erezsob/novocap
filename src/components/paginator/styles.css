@value defaultText, grayBg, blueDark from '../../common/variables.css';

.pages {
    list-style-type: none;
    display: flex;
    margin: 0;
    padding: 0;
}

.pages > li {
    background-color: white;
    color: defaultText;
}

.pages > li:first-child {
    border-radius: 4px 0 0 4px;
}

.pages > li:last-child {
    border-radius: 0 4px 4px 0;
}

.pages > li:nth-child(2n)  {
    background-color: grayBg;
}

.link {
    display: inline-block;
    width: 40px;
    height: 40px;
    font-size: 15px;
    text-align: center;
    line-height: 40px;
    position: relative;
}

.number {
    composes: link;
}

.disabled {
    pointer-events: none;
    opacity: 0.3;
}

.selected:after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 3px;
    background-color: blueDark;
}

.icon {
    position: relative;
    top: 6px;
}
