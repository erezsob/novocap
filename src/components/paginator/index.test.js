import React from 'react';
import { HashRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ArrowForward from 'material-ui-icons/ArrowForward';
import { Link } from 'react-router-dom';
import ArrowBack from 'material-ui-icons/ArrowBack';
import Paginator from './';
import styles from './styles.css';
import { DEFAULT_PAGER_SIZE } from './../../constants';

const props = {
  totalPages: 100,
  baseRoute: 'foo',
  pageSize: 20,
  currentPage: 3,
  className: 'bar',
  pagerSize: undefined
};

describe('<Paginator />', () => {
  let wrapper;
  let spyUpdatePagesArray;
  let spyComponentDidMount;
  let spyComponentWillReceiveProps;

  beforeEach(() => {
    spyComponentDidMount = jest.spyOn(Paginator.prototype, 'componentDidMount');
    spyComponentWillReceiveProps = jest.spyOn(
      Paginator.prototype,
      'componentWillReceiveProps'
    );
    spyUpdatePagesArray = jest.spyOn(Paginator.prototype, 'updatePagesArray');
    wrapper = shallow(<Paginator {...props} />);
  });

  afterEach(() => {
    spyUpdatePagesArray.mockClear();
    spyComponentDidMount.mockClear();
    spyComponentWillReceiveProps.mockClear();
  });

  it('renders', () => {
    expect(wrapper.find(`.${styles.pages}`)).toHaveLength(1);
  });

  it('renders properly', () => {
    const tree = renderer
      .create(
        <HashRouter>
          <Paginator {...props} />
        </HashRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with the class passed as className', () => {
    wrapper.setProps({ className: 'fooBar' });
    expect(wrapper.hasClass('fooBar')).toBeTruthy();
  });

  it('renders the right quantity of page numbers', () => {
    expect(wrapper.find(`.${styles.number}`)).toHaveLength(
      wrapper.state().pages.length
    );
  });

  describe('<ArrowBack />', () => {
    let previousPageLink;

    beforeEach(() => {
      previousPageLink = wrapper.childAt(0).find(Link);
    });

    it('renders', () => {
      wrapper.setProps({ currentPage: 2 });
      expect(wrapper.find(ArrowBack)).toHaveLength(1);
    });

    it('links to the right url', () => {
      expect(previousPageLink.prop('to')).toBe(
        `${props.baseRoute}?page=${props.currentPage - 1}&size=${
          props.pageSize
        }`
      );
    });

    it('has a `disabled` class if current page equals 1', () => {
      wrapper.setProps({ currentPage: 1 });
      previousPageLink = wrapper.childAt(0).find(Link);
      expect(previousPageLink.hasClass(`${styles.disabled}`)).toBeTruthy();
    });
  });

  describe('<ArrowForward />', () => {
    let nextPageLink;

    beforeEach(() => {
      nextPageLink = wrapper.childAt(7).find(Link);
    });

    it('renders', () => {
      expect(wrapper.find(ArrowForward)).toHaveLength(1);
    });

    it('links to the right url', () => {
      expect(nextPageLink.prop('to')).toBe(
        `${props.baseRoute}?page=${props.currentPage + 1}&size=${
          props.pageSize
        }`
      );
    });

    it('has a `disabled` class if current page equals last page', () => {
      wrapper.setProps({ currentPage: props.totalPages });
      nextPageLink = wrapper.childAt(7).find(Link);
      expect(nextPageLink.hasClass(`${styles.disabled}`)).toBeTruthy();
    });
  });

  it('assigns the default value of `DEFAULT_PAGER_SIZE` to pagerSize if it is undefined', () => {
    expect(wrapper.state().pages.length).toBe(DEFAULT_PAGER_SIZE);
  });

  it('assigns an array with the size of pagerSize to `pages`', () => {
    const wrapper = shallow(<Paginator {...props} pagerSize={12} />);
    expect(wrapper.state().pages.length).toBe(12);
  });

  it('creates an array with the right page numbers', () => {
    expect(wrapper.state().pages).toEqual([3, 4, 5, 6, 7, 8]);
    wrapper.setProps({ currentPage: 20 });
    expect(wrapper.state().pages).toEqual([20, 21, 22, 23, 24, 25]);
  });

  it('creates an array with the right page numbers if current page < 1', () => {
    wrapper.setProps({ currentPage: 0 });
    expect(wrapper.state().pages).toEqual([1, 2, 3, 4, 5, 6]);
  });

  it('creates an array with the right page numbers if current page > totalPages', () => {
    wrapper.setProps({ currentPage: 101 });
    expect(wrapper.state().pages).toEqual([95, 96, 97, 98, 99, 100]);
  });

  it('invokes updatePagesArray() with prop `currentPage`', () => {
    expect(spyUpdatePagesArray).toHaveBeenCalledWith(props.currentPage);
  });

  describe('Lifecycle methods', () => {
    describe('componentDidMount()', () => {
      it('is called when receiving new props', () => {
        expect(spyComponentDidMount).toHaveBeenCalledTimes(1);
      });
    });

    describe('componentWillReceiveProps()', () => {
      it('is called when receiving new props', () => {
        expect(spyComponentWillReceiveProps).toHaveBeenCalledTimes(0);
        wrapper.setProps({ currentPage: 5 });
        expect(spyComponentWillReceiveProps).toHaveBeenCalledTimes(1);
      });
    });
  });
});
