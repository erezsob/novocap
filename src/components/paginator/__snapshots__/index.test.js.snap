// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator /> renders properly 1`] = `
<ul
  className="pages bar"
>
  <li>
    <a
      className="link"
      href="#/foo?page=2&size=20"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1 icon"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </a>
  </li>
  <li>
    <a
      className="selected number"
      href="#/foo?page=3&size=20"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li>
    <a
      className="number"
      href="#/foo?page=4&size=20"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li>
    <a
      className="number"
      href="#/foo?page=5&size=20"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li>
    <a
      className="number"
      href="#/foo?page=6&size=20"
      onClick={[Function]}
    >
      6
    </a>
  </li>
  <li>
    <a
      className="number"
      href="#/foo?page=7&size=20"
      onClick={[Function]}
    >
      7
    </a>
  </li>
  <li>
    <a
      className="number"
      href="#/foo?page=8&size=20"
      onClick={[Function]}
    >
      8
    </a>
  </li>
  <li>
    <a
      className="link"
      href="#/foo?page=4&size=20"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1 icon"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
        />
      </svg>
    </a>
  </li>
</ul>
`;
